lib/validator.js

```diff
@@ -25,6 +25,26 @@ class Validator {
         return date instanceof Date
       }
     })
+
+    this.ajv.addKeyword({
+      keyword: 'isSet',
+      type: 'object',
+      errors: false,
+      validate: (type, data, schema) => {
+        if (data instanceof Set) {
+          data = Array.from(data)
+        }
+
+        if (schema.isSet) {
+          // undo what we do in convertSchemaToAjvFormat
+          // to prevent infinite recursive call
+          delete schema.isSet
+          schema.type = schema.type.filter(type => type !== 'object')
+        }
+
+        return this.ajv.validate(schema, data)
```

Comment:
I'm not sure that it will work with local references inside the `schema`.

Reply:
Do we have an example?

Reply:
```js
const schema = {
  anyOf: [
    { type: 'string' }
    {
      type: 'object',
      properties: {
        a: { type: 'array', items: { $ref: '#/anyOf/0' } }
      }
    }
  ]
}
```

Reply:
Nah It doesn't work :(. The `schema` arg in the validate function would just be `{ type: [ 'array' ], items: { '$ref': '#/anyOf/0' } }`.

Ugh let me think again. Please also let me know if you got any ideas.